### PR TITLE
fix/ perp curie errors

### PR DESCRIPTION
**Pull request by vic-en**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
https://github.com/hummingbot/hummingbot/issues/5693
This rp fixes some issues with perp connector

**Tests performed by the developer**:
- Did end-to-end testing between client and qa and ensure I don't have error.
- Ensured that unit tests pass.

**Tips for QA testing**:
- Basically do end-to-end testing and confirm issue doesn't persist.

### CODE REVIEW

**Comment by nikspz** [comment]
@vic-en Could you please resolve branch conflicts?

**Comment by nikspz** [comment]
Tests performed:

Cloned and installed fix branch
gateway generate certs and gateway create - confirmed working
Client testing: blocked with issue on spot-perpetual https://github.com/hummingbot/hummingbot/issues/5890
test endpoints
ping gateway endpoint - confirmed working
/amm/perp/market-prices - confirmed working
/amm/perp/market-status - confirmed working
/amm/perp/pairs - confirmed working
/amm/perp/position - confirmed working
/amm/perp/open - confirmed working
/amm/perp/close - confirmed working
/evm/approve - confirmed working

**Comment by vic-en** [comment]
@nikspz Resolved.

**Comment by rapcmia** [comment]
PR update:
- Setup long term bots for test monitoring 
- We have observed that there are no ``Unhandled error in background task: unsupported operand type(s) for -: 'NoneType' and 'decimal.Decimal'`` for the longest time 
- Noticed issues on order execution which is not related to this PR 
- There were instances of `'ServerDisconnectedError'` on both AWS and local tests
   - While this happened we review the docker log files (attached) and we'd like to confirm if this is possibly something to do with the default nodeURL `rpc.ankr.com/optimism` since its return an error of `https://links.ethers.org/v5-errors-CALL_EXCEPTION`?
```   
2022-11-16 10:44:59,004 - 75262 - hummingbot.core.gateway.gateway_http_client - WARNING - Call to https://localhost:31207/amm/perp/position failed. See logs for more details.
2022-11-16 10:44:59,007 - 75262 - GatewayEVMPerpetual - ERROR - 'ServerDisconnectedError' object has no attribute 'get'
Traceback (most recent call last):
  File "/home/ubuntu/5885/hummingbot/connector/gateway/amm/gateway_evm_perpetual.py", line 456, in _status_polling_loop
    await safe_gather(
  File "/home/ubuntu/5885/hummingbot/core/utils/async_utils.py", line 22, in safe_gather
    return await asyncio.gather(*args, **kwargs)
  File "/home/ubuntu/5885/hummingbot/connector/gateway/amm/gateway_evm_perpetual.py", line 555, in update_positions
    trading_pair = f"{position.get('base')}-{position.get('quote')}"
AttributeError: 'ServerDisconnectedError' object has no attribute 'get'
```

We also observed the error `'ValueError' object has no attribute 'get'` and `call....perp/position failed` after getting the `ServerDisconnectedError`
```
2022-11-18 18:11:38,107 - 3311 - hummingbot.core.gateway.gateway_http_client - WARNING - Call to https://localhost:26762/amm/perp/position failed. See logs for more details.
2022-11-18 18:11:38,107 - 3311 - GatewayEVMPerpetual - ERROR - 'ValueError' object has no attribute 'get'
Traceback (most recent call last):
  File "/home/rapcmia/github/5885/hummingbot/connector/gateway/amm/gateway_evm_perpetual.py", line 456, in _status_polling_loop
    await safe_gather(
  File "/home/rapcmia/github/5885/hummingbot/core/utils/async_utils.py", line 22, in safe_gather
    return await asyncio.gather(*args, **kwargs)
  File "/home/rapcmia/github/5885/hummingbot/connector/gateway/amm/gateway_evm_perpetual.py", line 555, in update_positions
    trading_pair = f"{position.get('base')}-{position.get('quote')}"
AttributeError: 'ValueError' object has no attribute 'get'
```
image

Pending:
- Running long term bot using optimism infura nodeURL `https://optimismmainnet.infura.io/v3/...` to compare test results

[5885.zip](https://github.com/hummingbot/hummingbot/files/10040660/5885.zip)

**Comment by JeremyKono** [comment]
@vic-en can you please resolve conflicts in this PR? Thanks

**Review by rapcmia — APPROVED** [review]
Test summary:
- Setup docker and source gateway successfully 
- Setup Spot-Perpetual Arbitrage using Perp as perpetual connector
- Ran long term bot and monitor tests
	- The issue on `NoneType' and 'decimal.Decimal` logs has been fixed
	- Order execution resulted on a different error not related to the PR, #5890 
	- There are still issues elated to `'ValueError' object is not subscriptable` that seems to be not related to the PR, #5902
- Ran endpoint test using cURL, all ok 
- Tested on both source and docker client build
